include/alpaka/mem/buf/BufGenericSycl.hpp

```diff
@@ -53,92 +54,102 @@ namespace alpaka
                 "The idx type of TExtent and the TIdx template parameter have to be identical!");
         }
 
-        TDev m_dev;
+        DevGenericSycl<TPltf> m_dev;
         Vec<TDim, TIdx> m_extentElements;
-        sycl::buffer<TElem, TDim::value> m_buffer;
+        std::shared_ptr<TElem> m_spMem;
     };
 } // namespace alpaka
 
 namespace alpaka::trait
 {
     //! The BufGenericSycl device type trait specialization.
-    template<typename TElem, typename TDim, typename TIdx, typename TDev>
-    struct DevType<BufGenericSycl<TElem, TDim, TIdx, TDev>>
+    template<typename TElem, typename TDim, typename TIdx, typename TPltf>
+    struct DevType<BufGenericSycl<TElem, TDim, TIdx, TPltf>>
     {
-        using type = TDev;
+        using type = DevGenericSycl<TPltf>;
     };
 
     //! The BufGenericSycl device get trait specialization.
-    template<typename TElem, typename TDim, typename TIdx, typename TDev>
-    struct GetDev<BufGenericSycl<TElem, TDim, TIdx, TDev>>
+    template<typename TElem, typename TDim, typename TIdx, typename TPltf>
+    struct GetDev<BufGenericSycl<TElem, TDim, TIdx, TPltf>>
     {
-        static auto getDev(BufGenericSycl<TElem, TDim, TIdx, TDev> const& buf)
+        ALPAKA_FN_HOST static auto getDev(BufGenericSycl<TElem, TDim, TIdx, TPltf> const& buf)
```

Comment:
Mh, not sure this is required since SYCL doesn't care about attributes like CUDA does. But this is an interface design question. What do you think @psychocoderHPC and @bernhardmgruber?

Reply:
A more general question: is there any backend that actually requires using `ALPAKA_FN_HOST` ?

AFAIK, `__host__` is optional with CUDA and HIP, because it's the default if no attribute is specified.
And the other backends defined it to nothing.

Reply:
I dislike `ALPAKA_FN_HOST` in all places. It just adds noise.